src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -453,17 +450,18 @@ private void updateOdometry() {
         lastModulePositions[moduleIndex] = modulePositions[moduleIndex];
       }
       if (hasNullModulePosition) {
-        if (!gyroInputs.connected || gyroInputs.odometryYawPositions[deltaIndex].get() == null) {
+        if (!gyroInputs.connected
+            || sample.values().get(new SignalID(SignalType.GYRO, -1)) == null) {
```

Comment:
Hardcoding of the gyro signal id has the potential to drift. If you're going to do it, I'd suggest a constant that gets reused.